src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs

```diff
@@ -430,12 +497,98 @@ internal static class Shards
     {
         #region common shard methods
 
+        private const string LeaseRetryTimer = "lease-retry";
+
         /// <summary>
-        /// TBD
+        /// Will call onLeaseAcquired when completed, also when lease isn't used
         /// </summary>
-        public static void Initialized<TShard>(this TShard shard) where TShard : IShard
+        /// <typeparam name="TShard"></typeparam>
+        /// <param name="shard"></param>
+        public static void AcquireLeaseIfNeeded<TShard>(this TShard shard) where TShard : IShard
+        {
+            if (shard.Lease != null)
+            {
+                shard.TryGetLease(shard.Lease);
+                shard.Context.Become(shard.AwaitingLease());
+            }
+            else
+                shard.OnLeaseAcquired();
+        }
+
+        public static void TryGetLease<TShard>(this TShard shard, Lease lease) where TShard : IShard
+        {
+            shard.Log.Info("Acquiring lease {0}", lease.Settings);
+
+            var self = shard.Self;
+            lease.Acquire(reason =>
+            {
+                self.Tell(new Shard.LeaseLost(reason));
```

Comment:
this callback should be called e.g. in following situation:
* we successfully acquire lease initially
* lease is doing internally some renewal of the lease
* because of e.g. network issue the renewal failed
* lease can send notification via the callback, that it was lost